stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/messages/adapter/view/MessageReplyView.kt

```diff
@@ -0,0 +1,163 @@
+import io.getstream.chat.android.ui.utils.extensions.getColorCompat
+
+public class MessageReplyView : FrameLayout {
+    private val binding: StreamUiMessageReplyViewBinding =
+        StreamUiMessageReplyViewBinding.inflate(LayoutInflater.from(context), this, true)
+
+    public constructor(context: Context) : super(context)
+    public constructor(context: Context, attrs: AttributeSet?) : super(context, attrs)
+    public constructor(context: Context, attrs: AttributeSet?, defStyleAttr: Int) : super(context, attrs, defStyleAttr)
+
+    public fun setMessage(message: Message, isMine: Boolean) {
+        setUserAvatar(message)
+        setAvatarPosition(isMine)
+        setReplyBackground(message, isMine)
+        setAttachmentImage(message)
+        setReplyText(message)
+    }
+
+    private fun setUserAvatar(message: Message) {
+        binding.avatarView.setUserData(message.user)
+    }
+
+    private fun setAvatarPosition(isMine: Boolean) {
+        binding.root.updateConstraints {
+            clear(R.id.avatarView, ConstraintSet.START)
+            clear(R.id.avatarView, ConstraintSet.END)
+            clear(R.id.replyContainer, ConstraintSet.END)
+            clear(R.id.replyContainer, ConstraintSet.END)
+        }
+        binding.avatarView.updateLayoutParams<ConstraintLayout.LayoutParams> {
+            if (isMine) {
+                endToEnd = ConstraintLayout.LayoutParams.PARENT_ID
+                startToEnd = R.id.replyContainer
+            } else {
+                startToStart = ConstraintLayout.LayoutParams.PARENT_ID
+                endToStart = R.id.replyContainer
+            }
+            marginStart = CONTENT_MARGIN
+            marginEnd = CONTENT_MARGIN
+        }
+        binding.replyContainer.updateLayoutParams<ConstraintLayout.LayoutParams> {
+            if (isMine) {
+                startToStart = ConstraintLayout.LayoutParams.PARENT_ID
+                endToStart = R.id.avatarView
+            } else {
+                startToEnd = R.id.avatarView
+                endToEnd = ConstraintLayout.LayoutParams.PARENT_ID
+            }
+            marginStart = CONTENT_MARGIN
+            marginEnd = CONTENT_MARGIN
+        }
+    }
+
+    private fun setReplyBackground(message: Message, isMine: Boolean) {
+        val shapeAppearanceModel = ShapeAppearanceModel.builder()
+            .setAllCornerSizes(REPLY_CORNER_RADIUS)
+            .setBottomLeftCornerSize(if (isMine.not()) 0f else REPLY_CORNER_RADIUS)
+            .setBottomRightCornerSize(if (isMine) 0f else REPLY_CORNER_RADIUS)
+            .build()
+        val isLink = message.attachments
+            .lastOrNull()
+            ?.type == ModelType.attach_link
+        binding.replyContainer.background = MaterialShapeDrawable(shapeAppearanceModel).apply {
+            when {
+                isLink -> {
+                    paintStyle = Paint.Style.FILL
+                    setTint(context.getColorCompat(R.color.stream_ui_blue_alice))
+                }
```

Comment:
Some of the design shows all attachments with this blue background (though some don't). Did we ask about the logic there?

Reply:
Asked designers, when original message contains link, then and only then we color background with light blue (The designs are already fixed in Figma).